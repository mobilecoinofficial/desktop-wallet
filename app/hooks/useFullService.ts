import { useContext } from 'react';

import FullServiceContext from '../contexts/FullServiceContext';

// TODO -- rename these files better
const useFullService = () => useContext(FullServiceContext);

export default useFullService;
