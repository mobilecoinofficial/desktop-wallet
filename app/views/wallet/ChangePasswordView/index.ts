import ChangePasswordView from './ChangePasswordView';

export default ChangePasswordView;
