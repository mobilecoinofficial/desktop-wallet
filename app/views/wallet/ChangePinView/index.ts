import ChangePinView from './ChangePinView';

export default ChangePinView;
