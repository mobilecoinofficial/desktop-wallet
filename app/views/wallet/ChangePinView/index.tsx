import React from 'react';
import type { FC } from 'react';

import {
  Box,
  Breadcrumbs,
  Container,
  FormHelperText,
  FormLabel,
  Link,
  Typography,
  makeStyles,
} from '@material-ui/core';
import { Formik, Form, Field } from 'formik';
import { TextField } from 'formik-material-ui';
import { useSnackbar } from 'notistack';
import { useTranslation } from 'react-i18next';
import { Link as RouterLink } from 'react-router-dom';
import * as Yup from 'yup';

import { SubmitButton } from '../../../components';
import routePaths from '../../../constants/routePaths';
import useIsMountedRef from '../../../hooks/useIsMountedRef';
import type { Theme } from '../../../theme';
import LocalStore from '../../../utils/LocalStore';
import { makeHash } from '../../../utils/hashing';

const useStyles = makeStyles((theme: Theme) => {
  return {
    button: {
      width: 200,
    },
    cardContainer: {
      paddingBottom: 64,
      paddingTop: 8 * 4,
    },
    cardContent: {
      display: 'flex',
      flexDirection: 'column',
      minHeight: 400,
      padding: theme.spacing(4),
    },
    center: {
      display: 'flex',
      justifyContent: 'center',
    },
    code: {
      alignItems: 'center',
      display: 'flex',
      flexDirection: 'column',
      letterSpacing: '.70rem',
      marginRight: '-.70rem',
      padding: theme.spacing(1),
    },
    form: {
      paddingBottom: theme.spacing(2),
    },
    label: {
      width: '100%',
    },
    modal: {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'center',
    },
    paper: {
      backgroundColor: theme.palette.background.paper,
      border: '2px solid #000',
      boxShadow: theme.shadows[5],
      padding: theme.spacing(2, 4, 3),
    },
    root: {
      backgroundColor: theme.palette.background.dark,
      display: 'flex',
      flexDirection: 'column',
      minHeight: '100vh',
    },
  };
});

const ChangePinView: FC = () => {
  const classes = useStyles();
  const { enqueueSnackbar } = useSnackbar();
  const isMountedRef = useIsMountedRef();
  const { t } = useTranslation('ChangePinView');

  return (
    <Container className={classes.cardContainer} maxWidth="sm">
      <Breadcrumbs separator=">" aria-label="breadcrumb">
        <Link color="inherit" to={routePaths.APP_SETTINGS} component={RouterLink}>
          <Typography color="textSecondary">{t('settingsBreadcrumb')}</Typography>
        </Link>
        <Typography color="textPrimary">{t('changePinBreadcrumb')}</Typography>
      </Breadcrumbs>
      <Box
        alignItems="center"
        display="flex"
        justifyContent="space-between"
        my={3}
        flexDirection="column"
      >
        <Typography variant="body2" color="textSecondary">
          {t('description')}
        </Typography>
        <br />
        <Typography variant="body2" color="textSecondary">
          {t('instructions')}
        </Typography>
      </Box>
      <Box flexGrow={1} mt={3}>
        <Formik
          initialValues={{
            newPin: '',
            newPinConfirmation: '',
            submit: null,
          }}
          validationSchema={Yup.object().shape({
            newPin: Yup.string()
              .min(6, t('PinMin'))
              .max(12, t('PinMax'))
              .required(t('PinRequired')),
            newPinConfirmation: Yup.string()
              .oneOf([Yup.ref('newPin')], t('PinConfirmationRef'))
              .required(t('PinConfirmationRequired')),
          })}
          onSubmit={(values, { setStatus, resetForm }) => {
            const hashedPin = makeHash(values.newPin);
            const localStore = new LocalStore();
            localStore.setHashedPin(String(hashedPin));
            if (isMountedRef.current) {
              enqueueSnackbar(t('enqueue'), {
                variant: 'success',
              });
            }
            setStatus({ success: true });
            resetForm();
          }}
        >
          {({ errors, isSubmitting, dirty, isValid, submitForm }) => {
            return (
              <Form>
                <Box pt={4}>
                  <FormLabel component="legend">
                    <Typography color="primary">{t('formLabel')}</Typography>
                  </FormLabel>
                  <Field
                    component={TextField}
                    fullWidth
                    label={t('newPinLabel')}
                    margin="normal"
                    name="newPin"
                    type="Pin"
                  />
                  <Field
                    component={TextField}
                    fullWidth
                    label={t('PinConfirmationLabel')}
                    margin="normal"
                    name="newPinConfirmation"
                    type="Pin"
                  />
                </Box>
                {errors.submit && (
                  <Box mt={3}>
                    <FormHelperText error>{errors.submit}</FormHelperText>
                  </Box>
                )}
                <SubmitButton
                  disabled={!dirty || !isValid || isSubmitting}
                  onClick={submitForm}
                  isSubmitting={isSubmitting}
                >
                  {t('changePinButton')}
                </SubmitButton>
              </Form>
            );
          }}
        </Formik>
      </Box>
    </Container>
  );
};

export default ChangePinView;
