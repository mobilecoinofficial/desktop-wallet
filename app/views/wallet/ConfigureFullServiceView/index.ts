import ConfigureFullServiceView from './ConfigureFullServiceView';

export default ConfigureFullServiceView;
