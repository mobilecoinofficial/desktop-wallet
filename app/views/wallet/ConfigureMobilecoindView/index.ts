import ConfigureMobilecoindView from './ConfigureMobilecoindView';

export default ConfigureMobilecoindView;
