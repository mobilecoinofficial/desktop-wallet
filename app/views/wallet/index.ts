export { default as ChangePasswordView } from './ChangePasswordView';
export { default as ChangePinView } from './ChangePinView';
export { default as ConfigureFullServiceView } from './ConfigureFullServiceView';
export { default as ContactsBookView } from './ContactsBookView';
export { default as DashboardView } from './DashboardView';
export { default as GiftingView } from './GiftingView';
export { default as HistoryView } from './HistoryView';
export { default as PrivacyPolicyView } from './PrivacyPolicyView';
export { default as RetrieveEntropyView } from './RetrieveEntropyView';
export { default as SettingsView } from './SettingsView';
export { default as TermsOfUseView } from './TermsOfUseView';
export { default as TransactionView } from './TransactionView';
export { default as TransactionDetailsView } from './TransactionDetailsView';
