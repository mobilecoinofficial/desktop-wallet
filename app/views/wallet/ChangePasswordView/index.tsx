import React from 'react';
import type { FC } from 'react';

import {
  Box,
  Breadcrumbs,
  Container,
  FormHelperText,
  FormLabel,
  Link,
  Typography,
  makeStyles,
} from '@material-ui/core';
import { Formik, Form, Field } from 'formik';
import { TextField } from 'formik-material-ui';
import { useSnackbar } from 'notistack';
import { Link as RouterLink } from 'react-router-dom';
import * as Yup from 'yup';

import { SubmitButton } from '../../../components';
import routePaths from '../../../constants/routePaths';
import useFullService from '../../../hooks/useFullService';
import useIsMountedRef from '../../../hooks/useIsMountedRef';
import type { Theme } from '../../../theme';

const useStyles = makeStyles((theme: Theme) => {
  return {
    button: {
      width: 200,
    },
    cardContainer: {
      paddingBottom: 64,
      paddingTop: 8 * 4,
    },
    cardContent: {
      display: 'flex',
      flexDirection: 'column',
      minHeight: 400,
      padding: theme.spacing(4),
    },
    center: {
      display: 'flex',
      justifyContent: 'center',
    },
    code: {
      alignItems: 'center',
      display: 'flex',
      flexDirection: 'column',
      letterSpacing: '.70rem',
      marginRight: '-.70rem',
      padding: theme.spacing(1),
    },
    form: {
      paddingBottom: theme.spacing(2),
    },
    label: {
      width: '100%',
    },
    modal: {
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'center',
    },
    paper: {
      backgroundColor: theme.palette.background.paper,
      border: '2px solid #000',
      boxShadow: theme.shadows[5],
      padding: theme.spacing(2, 4, 3),
    },
    root: {
      backgroundColor: theme.palette.background.dark,
      display: 'flex',
      flexDirection: 'column',
      minHeight: '100vh',
    },
  };
});

const ChangePasswordView: FC = () => {
  const classes = useStyles();
  const { enqueueSnackbar } = useSnackbar();
  const isMountedRef = useIsMountedRef();
  const { changePassword } = useFullService();

  return (
    <Container className={classes.cardContainer} maxWidth="sm">
      <Breadcrumbs separator=">" aria-label="breadcrumb">
        <Link
          color="inherit"
          to={routePaths.APP_SETTINGS}
          component={RouterLink}
        >
          <Typography color="textSecondary">Settings</Typography>
        </Link>
        <Typography color="textPrimary">Change Password</Typography>
      </Breadcrumbs>
      <Box
        alignItems="center"
        display="flex"
        justifyContent="space-between"
        my={3}
        flexDirection="column"
      >
        <Typography variant="body2" color="textSecondary">
          Changing your password will change how we encrypt your secret Entropy.
          You should always use strong (unguessable) passwords and change them
          regularly.
        </Typography>
        <br />
        <Typography variant="body2" color="textSecondary">
          To change your current password, please enter your current password
          and your new password.
        </Typography>
      </Box>
      <Box flexGrow={1} mt={3}>
        <Formik
          initialValues={{
            newPassword: '',
            newPasswordConfirmation: '',
            oldPassword: '',
            submit: null,
          }}
          validationSchema={Yup.object().shape({
            newPassword: Yup.string()
              .min(8, 'New Password must be at least 8 characters in length.')
              .max(99, 'Passwords cannot be more than 99 characters.')
              .required('New Password is required'),
            newPasswordConfirmation: Yup.string()
              .oneOf([Yup.ref('newPassword')], 'Must match New Password')
              .required('New Password Confirmation is required'),
          })}
          isInitialValid={false}
          onSubmit={async (
            values,
            {
              setErrors, setStatus, setSubmitting, resetForm,
            },
          ) => {
            try {
              setSubmitting(true);
              await changePassword(values.oldPassword, values.newPassword);
              if (isMountedRef.current) {
                enqueueSnackbar('Successfully change password!', {
                  variant: 'success',
                });
                setStatus({ success: true });
                setSubmitting(false);
                resetForm();
              }
            } catch (err) {
              if (isMountedRef.current) {
                setStatus({ success: false });
                setErrors({ submit: err.message });
                setSubmitting(false);
              }
            }
          }}
        >
          {({
            errors, isSubmitting, isValid, submitForm,
          }) => {
            return (
              <Form>
                <Box pt={4}>
                  <FormLabel component="legend">
                    <Typography color="primary">Change Password</Typography>
                  </FormLabel>
                  <Field
                    component={TextField}
                    fullWidth
                    label="Old Password"
                    margin="normal"
                    name="oldPassword"
                    type="password"
                  />
                  <Field
                    component={TextField}
                    fullWidth
                    label="New Password"
                    margin="normal"
                    name="newPassword"
                    type="password"
                  />
                  <Field
                    component={TextField}
                    fullWidth
                    label="Confirm New Password"
                    margin="normal"
                    name="newPasswordConfirmation"
                    type="password"
                  />
                </Box>
                {errors.submit && (
                  <Box mt={3}>
                    <FormHelperText error>{errors.submit}</FormHelperText>
                  </Box>
                )}
                <SubmitButton
                  disabled={!isValid || isSubmitting}
                  onClick={submitForm}
                  isSubmitting={isSubmitting}
                >
                  Change Password
                </SubmitButton>
              </Form>
            );
          }}
        </Formik>
      </Box>
    </Container>
  );
};

export default ChangePasswordView;
