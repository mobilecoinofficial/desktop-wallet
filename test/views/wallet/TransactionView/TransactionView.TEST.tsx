import React from 'react';

import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import useFullService from '../../../../app/hooks/useFullService';
import { TransactionView } from '../../../../app/views/wallet';
import renderSnapshot from '../../../renderSnapshot';

jest.mock('../../../../app/hooks/useFullService');

function setupComponent() {
  const mockUseFullService = useFullService as jest.MockedFunction<typeof useFullService>;

  // @ts-ignore mock
  mockUseFullService.mockImplementation(() => ({
    accountName: 'account name',
    b58Code: 'b58 code',
  }));

  const { asFragment } = renderSnapshot(<TransactionView />);

  const sendQuery = screen.queryByText(
    'Please enter the amount of MOB you want to send and the public address of the recipient.'
  );
  const receiveQuery = screen.queryByText(
    'To receive MOB, you must share your public address code to the sender.'
  );

  return {
    asFragment,
    receiveQuery,
    sendQuery,
  };
}

describe('TransactionView', () => {
  describe('component', () => {
    describe('render', () => {
      test('TransactionView renders correctly', () => {
        const { asFragment } = setupComponent();
        expect(asFragment()).toMatchSnapshot();
      });

      test('SendMobPanel renders correctly', () => {
        const { sendQuery, receiveQuery } = setupComponent();
        userEvent.click(screen.getByText('Send MOB'));
        expect(sendQuery).toBeInTheDocument();
        expect(receiveQuery).not.toBeInTheDocument();
      });

      test('ReceiveMobPanel renders correctly', async () => {
        const { sendQuery } = setupComponent();
        userEvent.click(screen.getByText('Receive MOB'));
        await waitFor(() =>
          expect(
            screen.queryByText(
              /To receive MOB, you must share your public address code to the sender./i
            )
          ).toBeInTheDocument()
        );
        expect(sendQuery).not.toBeInTheDocument();
      });
    });
  });
});
