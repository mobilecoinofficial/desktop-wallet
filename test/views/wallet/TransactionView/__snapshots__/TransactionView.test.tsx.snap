// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionView component render TransactionView renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root makeStyles-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            style="overflow: hidden;"
          >
            <div
              class="MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary MuiTab-selected MuiTab-fullWidth"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  Send MOB
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary MuiTab-fullWidth"
                role="tab"
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  Receive MOB
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="PrivateTabIndicator-root PrivateTabIndicator-colorSecondary MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root"
        >
          <div
            class="MuiContainer-root makeStyles-cardContainer MuiContainer-maxWidthSm"
          >
            <div
              class="MuiBox-root MuiBox-root"
            >
              <div
                class="MuiBox-root MuiBox-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                >
                  Please enter the amount of MOB you want to send and the public address of the recipient.
                </p>
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root"
            >
              <form
                action="#"
              >
                <div
                  class="MuiBox-root MuiBox-root"
                >
                  <legend
                    class="MuiFormLabel-root makeStyles-form"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                    >
                      Select Account
                    </p>
                  </legend>
                  <div
                    class="MuiFormGroup-root"
                    role="radiogroup"
                  >
                    <div
                      class="MuiBox-root MuiBox-root"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
                      >
                        Account Name
                      </p>
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
                      >
                        Account Balance
                      </p>
                    </div>
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked MuiRadio-checked MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            checked=""
                            class="PrivateSwitchBase-input"
                            name="senderPublicAddress"
                            type="radio"
                            value="b58 code"
                          />
                          <div
                            class="PrivateRadioButtonIcon-root PrivateRadioButtonIcon-checked"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                              />
                            </svg>
                          </div>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label makeStyles-label MuiTypography-body1"
                      >
                        <div
                          class="MuiBox-root MuiBox-root"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
                          >
                             account name:
                            <span
                              class="MuiBox-root MuiBox-root"
                              data-testid="short-code"
                            >
                              <span
                                class="MuiBox-root MuiBox-root makeStyles-lowercased"
                              >
                                b
                              </span>
                              <span
                                class="MuiBox-root MuiBox-root makeStyles-number"
                              >
                                5
                              </span>
                              <span
                                class="MuiBox-root MuiBox-root makeStyles-number"
                              >
                                8
                              </span>
                              <span
                                class="MuiBox-root MuiBox-root makeStyles-number"
                              >

                              </span>
                              <span
                                class="MuiBox-root MuiBox-root makeStyles-lowercased"
                              >
                                -
                              </span>
                              <span
                                class="MuiBox-root MuiBox-root makeStyles-lowercased"
                              >
                                c
                              </span>
                              <span
                                class="MuiBox-root MuiBox-root makeStyles-lowercased"
                              >
                                o
                              </span>
                              <span
                                class="MuiBox-root MuiBox-root makeStyles-lowercased"
                              >
                                d
                              </span>
                              <span
                                class="MuiBox-root MuiBox-root makeStyles-lowercased"
                              >
                                e
                              </span>
                            </span>
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
                          >
                            <span>
                              0.000088888888
                            </span>
                          </p>
                        </div>
                      </span>
                    </label>
                  </div>
                </div>
                <div
                  class="MuiBox-root MuiBox-root"
                >
                  <legend
                    class="MuiFormLabel-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                    >
                      Transaction Details
                    </p>
                  </legend>
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                      data-shrink="false"
                    >
                      Recipient Public Address
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        name="recipientPublicAddress"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                      data-shrink="true"
                      for="mobAmount"
                      id="mobAmount-label"
                    >
                      MOB Amount
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                    >
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionStart"
                      >
                        <svg
                          height="20"
                          viewBox="0 0 125 125"
                          width="20"
                        >
                          <path
                            d="M62.362 124.724C27.975 124.724 0 96.748 0 62.364 0 27.977 27.975 0 62.362 0c34.388 0 62.362 27.977 62.362 62.364 0 34.385-27.975 62.36-62.362 62.36zm30.194-75.406a8.126 8.126 0 01-14.652 4.845H10.33a53.064 53.064 0 00-.636 8.201c0 29.041 23.625 52.669 52.667 52.669 21.268 0 39.632-12.672 47.938-30.859h-62.21a8.125 8.125 0 01-6.658 3.462 8.125 8.125 0 01-8.126-8.127 8.127 8.127 0 0114.512-5.027v-.003h65.808a52.607 52.607 0 001.407-12.114c0-29.043-23.628-52.671-52.67-52.671-22.763 0-42.199 14.516-49.54 34.777h65.082a8.126 8.126 0 0114.652 4.847z"
                            fill="#A6AAB4"
                          />
                        </svg>
                      </div>
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                        id="mobAmount"
                        inputmode="numeric"
                        type="text"
                        value="0.000000000000"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root MuiBox-root makeStyles-root"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disabled MuiButton-fullWidth MuiButtonBase-disabled"
                    data-testid="submit-button"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Send Payment
                    </span>
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
